### DISCUSSION TITLE
Dropping numpy array support for TableVectorizer

### DISCUSSION THREAD

**Discussion by Vincent-Maladiere** [skrub-data/skrub#816] [General]
Should we stop supporting numpy array inputs for TableVectorizer?
- From a data scientist's UX point of view, numpy arrays are seldom used as inputs to stats or ML models. Data scientists mainly use dataframes, from fetching data to writing data.
- Conceptually speaking, `TableVectorizer` deals with columns of different dtypes, while a numpy array only has one.
- This would allow us to avoid dealing with column indices instead of column names.
- This would also enable long-term laziness and support for different dataframes.

WDYT?

**Comment by GaelVaroquaux** [comment]
Let's do it for TableVectorizer (but with a clean error for inputs that are not understood)

**Comment by jeromedockes** [comment]
this has been fixed by #902 -- if the input is a numpy array a warning is issued and it is converted to a dataframe
